GitUI/BranchTreePanel/BaseRefTree.cs

```diff
@@ -80,5 +82,69 @@ internal override void UpdateVisibility()
 
             base.UpdateVisibility();
         }
+
+        protected IEnumerable<IGitRef> PrioritizedBranches(IReadOnlyList<IGitRef> branches)
+        {
+            return OrderByPriority(branches, node => node.LocalName, AppSettings.PrioritizedBranchNames);
+        }
+
+        protected IEnumerable<RemoteRepoNode> PrioritizedRemotes(IReadOnlyList<RemoteRepoNode> remotes)
+        {
+            return OrderByPriority(remotes.OrderBy(node => node.FullPath).ToList(), node => node.FullPath, AppSettings.PrioritizedRemoteNames);
+        }
+
+        /// <summary>
+        /// Order the references by the priorities in the setting regex
+        /// </summary>
+        /// <typeparam name="T">The type to prioritize, e.g. IGitRef.</typeparam>
+        /// <param name="references">The branches or remotes to prioritize.</param>
+        /// <param name="keySelector">Function in T to get the sorter.</param>
+        /// <param name="setting">String with regexes with priorities separated by semicolon.</param>
+        /// <returns>The resorted references.</returns>
+        private IEnumerable<T> OrderByPriority<T>(IReadOnlyList<T> references, Func<T, string> keySelector, string setting)
+        {
+            // Sort prio branches first (if set) with the compile cache (no need to instantiate)
+            string[] regexes = setting.Split(";", StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.TrimEntries)
+                .Select(regex => $"^({regex})$")
+                .ToArray();
+
+            if (!regexes.Any())
+            {
+                return references;
+            }
+
+            // A lot of regexes will push out entries from the static regex cache, increasing the time a lot (several hundred ms).
+            // (Switching the regex to the outer loop will decrease the effect but the code structure is simpler this way).
+            // The static .NET regex cache must be able to include at least all regexes in the loop,
+            // ideally also all common use in a "refresh" loop, so increase the size.
+            // A few usages in branches vs remote in this method, CommitInfo adds usage for
+            // split length of PrioritizedBranchNames (for remotes) and PrioritizedRemoteNames,
+            // a few usages in submodule status processing etc.
+            const int additionalRegexCacheEntries = 6;
```

Comment:
Why so sparing? Are cache entries such expensive?

Reply:
Probably not expensive, but 6+regex.Length is likely sufficient.
CommitInfo is maybe a better place to check this, or some other centralized init (at startup should be sufficient.). 
I guess the cost is when there additional usages without changing the formula.

Can change it to 10+regex.Length with a note that this should be done at startup.